SeaBattle/Sea/Sea.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Drawing;
+namespace SeaBattle
+{
+    public class Sea
+    {
+        private List<BaseShip> Ships { get; set; }
+        public int SeaWidth;
+        public int SeaHeight;
+        public Sea(int x, int y)
+        {
+            SeaWidth= x;
+            SeaHeight= y;
+            Ships = new List<BaseShip>();
+        }
+
+        // Property to provide indexer
+        public BaseShip this[int i]
+        {
+            get
+            {
+                if (i >= 0 && i < Ships.Count)
+                    return Ships[i];
+                else
+                    throw new IndexOutOfRangeException();
+            }
+            // checking value is not null, checking value is in BaseShip family, checking Sea[i] is present
+            set
+            {
+                if (value != null)
+                    if (value is BaseShip)
+                        if (i > 0 && i < Ships.Count)
+                            Ships[i] = value;
+                        else
+                            throw new Exception("Unable to add ship to the list");
+            }
+        }
+        // This function adds s1(BaseShip) to the Sea
+        // 1. + Checking if Point within Sea borders
+        // 2. + Checking if Point is available 
+        public void AddShip(BaseShip s1)
+        {
+            List<Point> NewLocations = new List<Point>();
+            foreach (var deck in s1.Decks)
+            {
+                NewLocations.Add(new Point(deck.Location.X, deck.Location.Y));
+
+                if (deck.Location.X > SeaWidth || deck.Location.Y > SeaHeight || deck.Location.X <= 0 || deck.Location.Y <= 0)
+
+                    throw new Exception("Ship can not exsist with this Sea borders");
+            }
+
+            if (ShipSpaceCheck(NewLocations))
+
+                Ships.Add(s1);
+  
+        }
+        public bool ShipSpaceCheck(List<Point> DecksPoints)
+        {
+            foreach (var deck in DecksPoints)
```

Comment:
Yes, with new namings it's much better, but still, a lot of nested loops, simplify

Reply:
so I need to use LINQ instead of loops and the ternary conditional operator instead of IF or? cant clearly understand but will try.

Reply:
Check now please, guess it much readable now.

Reply:
Just realised that I dont need new point and can do just (deck.X, deck.Y). What about improvement? You want LINQ or ? Btw look on IsShipLine method and say is it good?(baseShip)

Reply:
You can just pass deck.Location which is a Point type, no need to do anything with this data. Again, you can use linq if you wish, but not in the way you did before)